fetch-in-react/my-app/src/Users.tsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserDetails } from './UserDetails';
+import { UserCard } from './UserCard';
+import './Users.css';
+
+export type User = {
+  id: number;
+  name: string;
+  email: string;
+  company: {
+    name: string;
+  };
+};
+export function Users() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [users, setUsers] = useState<User[]>([]);
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+  useEffect(() => {
+    async function getUsers() {
+      try {
+        const res = await fetch('https://jsonplaceholder.typicode.com/users');
+        if (!res.ok) {
+          throw new Error(`fetch failed ${res.status}`);
+        }
+        const newusers = await res.json();
+        setUsers(newusers);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
+      }
+    }
+    getUsers();
+    return () => setIsLoading(false);
```

Comment:
Same here. Also, by making `isLoading` a dependency you run the risk of infinite re-renders. Note that you don't actually use `isLoading` anywhere in your effect function, so it's not really a dependency, anyway.